File: Selection Sort/Selection Sort.py
# This code is contributed by @iuliagroza

# Time Complexity: O(n ^ 2)
# Additional Space: O(1)

def selectionSort(a):
    # Traverse through all array elements
    for i in range(len(a)):
        # Find the minimum element in remaining unsorted array
        min_index = i
        for j in range(i + 1, len(a)):
            if a[min_index] > a[j]:
                min_index = j

        # Swap the found minimum element with the first element
        a[i], a[min_index] = a[min_index], a[i]

    for i in range(len(a)):
        print("%d" % a[i]),

a = [64, 25, 12, 22, 11]
selectionSort(a)

File: Selection Sort/Selection Sort.cpp
/// This code is contributed by @iuliagroza

/// Time Complexity: O(n^2)
/// Additional Space: O(1)

#include <iostream>
#include <algorithm>

void selectionSort(int a[], int n)
{
    for(int i=0; i<n-1; i++) {
        int min_index = i; // Minimum value position
        for(int j=i+1; j<n; j++) {
            if (a[min_index] > a[j]) {
                min_index = j;
            }
        std::swap(a[i], a[min_index]);
        }
    }
}

int main()
{
    std::ios_base::sync_with_stdio(0);
    std::cin.tie(0);
    std::cout.tie(0);

    int n, a[100];
    std::cin >> n;
    for(int i=0; i<n; i++) {
        std::cin >> a[i];
    }

    selectionSort(a, n);

    for(int i=0; i<n; i++) {
        std::cout << a[i] << ' ';
    }
    return 0;
}
